Assets/HoloToolkit/Utilities/Editor/ConfigureMenu.cs

```diff
@@ -22,34 +23,186 @@ public static void ShowHelp()
             Application.OpenURL("https://developer.microsoft.com/en-us/windows/holographic/unity_development_overview#Configuring_a_Unity_project_for_HoloLens");
         }
 
-        /// <summary>
-        /// Applies recommended scene settings to the current scenes
-        /// </summary>
-        [MenuItem("HoloToolkit/Configure/Apply HoloLens Scene Settings", priority = 0)]
-        public static void ApplySceneSettings()
+        [MenuItem("HoloToolkit/Configure/Apply All HoloLens Settings", priority = 0)]
+        public static void ApplyAllSettings()
         {
+            var output = new StringBuilder();
+            ApplySceneSettingsInternal(output);
+
+
+            var reload = ApplyProjectSettingsInternal(output);
+
+            EditorUtility.DisplayDialog("Results", output.ToString(), "Ok");
+
+            if (reload)
+            {
+                PromptForReload();
+            }
+        }
+
+
+        private static void ApplySceneSettingsInternal(StringBuilder output)
+        {
+            //todo if we find more than one camera, hololens will always use MainCamera (should)
+            //additional cameras could be used from a RenderTexture, etc. Should we warn?
+
+            output.AppendLine("** Checking Camera Settings");
+            Camera[] cameras = new Camera[Camera.allCamerasCount];
+            Camera.GetAllCameras(cameras);
+
             if (Camera.main == null)
             {
-                Debug.LogWarning(@"Could not apply settings - no camera tagged with ""MainCamera""");
+                output.AppendLine(@"Could not apply camera settings - no camera tagged with ""MainCamera""");
                 return;
             }
 
-            Camera.main.transform.position = Vector3.zero;
-            Camera.main.clearFlags = CameraClearFlags.SolidColor;
-            Camera.main.backgroundColor = Color.black;
-            Camera.main.nearClipPlane = 0.85f;
+            SetCameraDefaults(Camera.main, true, output);
+
+            //All done with MainCamera
+            cameras = cameras.Where(o => o != Camera.main).ToArray();
```

Comment:
I'm hesitant to replace anything in the scene and a prefab is a bit of an
overkill. One idea may be to replace it and then the original is disabled.
In that case it seems messy though. One issue is people take existing
scenes from their Unity projects and then bring them to the HoloLens. In
this case they have existing projects. Maybe they have code that looks for
the main camera position (by name or tag) and now we've replaced that. Its
a fringe case but entirely possible. Its easy enough to change the camera
and tell them about it. The settings aren't major (and they may have
already set their clip plane to be custom ie !=0.85f and we are just fixing
say, the position for them. I think telling them what we've changed is the
friendlier way.

On Wed, Jul 27, 2016 at 9:59 PM, Josh Wittner <notifications@github.com>
wrote:

> In Assets/HoloToolkit/Utilities/Editor/ConfigureMenu.cs
> <https://github.com/Microsoft/HoloToolkit-Unity/pull/134#discussion_r72565735>
> :
>
> >                  return;
> >              }
> >
> > -            Camera.main.transform.position = Vector3.zero;
> > -            Camera.main.clearFlags = CameraClearFlags.SolidColor;
> > -            Camera.main.backgroundColor = Color.black;
> > -            Camera.main.nearClipPlane = 0.85f;
> > +            SetCameraDefaults(Camera.main, true, output);
> > +
> > +            //All done with MainCamera
> > +            cameras = cameras.Where(o => o != Camera.main).ToArray();
>
> What if the tooling replaced the main camera with the prefab when this
> ran? Could keep the same transform as the previous camera, but first setup
> would have things that just worked with like two clicks.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Microsoft/HoloToolkit-Unity/pull/134/files/3933b816638f8cce7b4dc9c87ac4fbcaadee4ded#r72565735>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABqPDRn2UZ0GwRyGq7RKiYuotBmH7za4ks5qaDc1gaJpZM4JTeQs>
> .
>